tools/SetupFlow/DevHome.SetupFlow/ViewModels/CloneRepoSummaryInformationFileViewModel.cs

```diff
@@ -0,0 +1,229 @@
+    {
+        get
+        {
+            var repoPath = Path.Join(OwningAccount, RepoName);
+            return _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsDescription, FileName, repoPath);
+        }
+    }
+
+    public string ViewFileMessage { get; } = string.Empty;
+
+    public string RunFileMessage { get; } = string.Empty;
+
+    public CloneRepoSummaryInformationViewModel(SetupFlowOrchestrator setupFlowOrchestrator, ISetupFlowStringResource stringResource)
+    {
+        _relatedActivityId = setupFlowOrchestrator.ActivityId;
+        _stringResource = stringResource;
+        FileFoundMessage = _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsFileFound);
+        ViewFileMessage = _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsView);
+        RunFileMessage = _stringResource.GetLocalized(StringResourceKey.CloneRepoNextStepsRun);
+    }
+
+    [RelayCommand]
+    public void OpenFileInExplorer()
+    {
+        TelemetryFactory.Get<ITelemetry>().Log(
+            _eventName,
+            LogLevel.Critical,
+            new CloneRepoNextStepEvent(_viewOperation, RepoName),
+            _relatedActivityId);
+
+        if (FilePathAndName is null)
+        {
+            TelemetryFactory.Get<ITelemetry>().Log(
+                _eventName,
+                LogLevel.Critical,
+                new CloneRepoNextStepError(_viewOperation, new ArgumentNullException(nameof(FilePathAndName)).ToString(), RepoName),
+                _relatedActivityId);
+
+            _log.Warning("CloneRepoSummaryInformationViewModel", $"{nameof(FilePathAndName)} is null when trying to view file.");
+            return;
+        }
+
+        if (string.IsNullOrEmpty(FilePathAndName))
+        {
+            TelemetryFactory.Get<ITelemetry>().Log(
+                _eventName,
+                LogLevel.Critical,
+                new CloneRepoNextStepError(_viewOperation, new ArgumentException($"{nameof(FilePathAndName)} is empty.").ToString(), RepoName),
+                _relatedActivityId);
+
+            _log.Warning("CloneRepoSummaryInformationViewModel", $"{nameof(FilePathAndName)} is empty when trying to view file.");
+            return;
+        }
+
+        var processStartInfo = new ProcessStartInfo();
+        processStartInfo.UseShellExecute = true;
+
+        try
+        {
+            processStartInfo.FileName = Path.GetDirectoryName(FilePathAndName);
+        }
+        catch (ArgumentException ex)
+        {
+            TelemetryFactory.Get<ITelemetry>().Log(
+            _eventName,
+            LogLevel.Critical,
+            new CloneRepoNextStepError(_viewOperation, ex.ToString(), RepoName),
+            _relatedActivityId);
+
+            _log.Warning("CloneRepoSummaryInformationViewModel", $"{nameof(FilePathAndName)} is either empty, contains only white spaces, or contains invalid characters.");
+            return;
+        }
+        catch (PathTooLongException ex)

```

Comment:
>PathTooLongException [](http://example.com/codeflow?start=15&length=20)

why does this go to telemetry?  shouldn't this be told to the user?  isn't this user error?  doesn't the user need to fix this?

Reply:
Very true.  Removed because, if this file is too deep in the directory tree, the exception would have been caught in the loading screen and the repo would not be cloned.